tests/integration-tests/tests/common/osu_common.py

```diff
@@ -104,6 +105,60 @@ def run_individual_osu_benchmark(
     return job_id, output
 
 
+def results_csv_to_dict() -> dict:
+    filename = OSU_COMMON_DATADIR / "reference_results.csv"
+    dict_result = {}
+    with open(filename, newline="", encoding="utf-8-sig") as results:
+        reader = csv.DictReader(results, delimiter=";")
+        for row in reader:
+            mpi_version = row["mpi_version"]
+            test_name = row["test_name"]
+            packet_size = row["packet_size"]
+            if not dict_result.get(mpi_version):
+                dict_result[mpi_version] = {}
+            if not dict_result[mpi_version].get(test_name):
+                dict_result[mpi_version][test_name] = {}
+            if not dict_result[mpi_version][test_name].get(packet_size):
+                dict_result[mpi_version][test_name][packet_size] = {}
+
+            dict_result[mpi_version][test_name][packet_size]["average"] = float(
+                row["average"].replace(".", "").replace(",", ".")
+            )
+            dict_result[mpi_version][test_name][packet_size]["std"] = float(
+                row["std"].replace(".", "").replace(",", ".")
+            )
+    return dict_result
+
+
+def check_threshold(reference_results, metric_data, variability_factor=5) -> list:
+    check_errors = []
+    for metric in metric_data:
+        if metric["MetricName"] == "Latency":
+            for d in metric["Dimensions"]:
+                if d["Name"] == "PacketSize":
+                    packet_size = str(d["Value"])
+                elif d["Name"] == "OsuBenchmarkName":
+                    test_name = d["Value"]
+                elif d["Name"] == "MpiVariant":
+                    mpi_variant = d["Value"]
+        try:
+            threshold = reference_results[mpi_variant][test_name][packet_size]["average"] + (
+                variability_factor * reference_results[mpi_variant][test_name][packet_size]["std"]
+            )
+        except KeyError:
```

Comment:
if you want to avoid KeyError you can use `d.get("key")` rather than `d["key"]`.